### PR TITLE
Complexes pr 3

### PR DESCRIPTION
**Pull request by mikestillman**
This pull request includes the next set of minor package changes to get them working with Complexes (the current plan is to make this the default in the April 2025 M2 release, removing the old ChainComplex type from the Core).

Pinged authors: do these changes look ok to you? (They should all be minor: changing ChainComplex to Complex, etc)
- `KustinMiller`: @jankoboehm
- `EliminationMatrices`: @lrtbuse
- `EagonResolutions`: @eisenbud 
- `QuillenSuslin`: @bbarwick @bstone 
- `ExteriorIdeals`, `ExteriorModules`: @lucamata 
- `BettiCharacters`, `HighestWeights`: @galettof 

After this pull request, there are not too many packages left to transfer to work directly with the new `Complexes` package.

### CODE REVIEW

File: M2/Macaulay2/packages/NormalToricVarieties.m2 [hunk 1]
```diff
@@ -27,7 +27,7 @@ newPackage(
     Headline => "routines for working with normal toric varieties and related objects",
     Keywords => {"Toric Geometry"},
     PackageExports => {"Polyhedra", "Schubert2", "Varieties","Truncations"},
-    PackageImports => {"FourierMotzkin", "LLLBases"},
+    PackageImports => {"FourierMotzkin","Normaliz","LLLBases","Complexes"},
```
Comment (mahrud) [inline M2/Macaulay2/packages/NormalToricVarieties.m2:30]: NormalToricVarieties doesn't use Normaliz.

**Comment by bbarwick** [comment]
@mikestillman (cc: @bstone) The QuillenSuslin.m2 changes look good to me. Thanks!

**Comment by lrtbuse** [comment]
Dear Mike,

Changes in EliminationMatrices seem good to me; thank you very much for taking care of that. 

Best,
Laurent.

**Comment by galettof** [comment]
Hi Mike,

From what I can tell, the changes are fine for the HighestWeights and BettiCharacters packages.

Thank you!

**Comment by mikestillman** [comment]
@ggsmith `K3Carpets` is failing: it is using `CompleteIntersectionResolutions`.  Unfortunately, it doesn't look as trivial to modify as the other packages we have done.  It uses also `NonminimalComplexes`, which we have not yet modified to work with Complexes.

**Comment by ggsmith** [comment]
> Do you want to revert this PR to commit [cfe3b5d](https://github.com/Macaulay2/M2/commit/cfe3b5df5611c250e83022df063f97786c37f35b), which passed all tests? I think changes to every package up to that point except for KustinMiller and EagonResolutions have been approved by the authors.
> 
> The remainder can be added in the next PR perhaps.

Not yet—we're stilling working on it.

**Comment by jankoboehm** [comment]
Hi Mike, The changes look very good with respect to the KustinMiller package, provided the tests pass of course. Thanks a lot!

**Comment by mikestillman** [comment]
@ggsmith I'm not sure I can find the bug in time, before the release gets made.  We might need to back off to the last pr-3 pull request...

**Comment by mikestillman** [comment]
@d-torrance What do you think about this?  I'm also concerned that I won't be able to quickly fix the conflict errors after David's PR for PencilsOfQuadrics...

**Comment by d-torrance** [comment]
Yeah, holding off on this PR until after the release sounds like a good idea.

**Comment by mahrud** [comment]
In the current version `rawMinimalBetti` receives a raw computation that hasn't started yet, so here is a fix:
```diff
diff --git a/M2/Macaulay2/packages/Complexes/FreeResolutions.m2 b/M2/Macaulay2/packages/Complexes/FreeResolutions.m2
index 27f0bd9aa8..a472fffe0f 100644
--- a/M2/Macaulay2/packages/Complexes/FreeResolutions.m2
+++ b/M2/Macaulay2/packages/Complexes/FreeResolutions.m2
@@ -202,6 +202,7 @@ resolutionObjectInEngine = (opts, M, matM) -> (
         complex maps
         );

+    if not opts.StopBeforeComputation then
     RO.compute(opts.LengthLimit, opts.DegreeLimit);
     RO.complex(opts.LengthLimit)
     )
@@ -691,7 +692,8 @@ minimalBetti Module := BettiTally => opts -> M -> (
        nvars := # generators(R, CoefficientRing => A);
        lengthlimit = nvars + if A === ZZ then 1 else 0;
         );
-    C = freeResolution(M, DegreeLimit => degreelimit, LengthLimit => lengthlimit + 1, Strategy => Nonminimal);
+    C = freeResolution(M, DegreeLimit => degreelimit, LengthLimit => lengthlimit + 1,
+       Strategy => Nonminimal, StopBeforeComputation => true);
     rC := M.cache.ResolutionObject.RawComputation;
     B := unpackEngineBetti rawMinimalBetti(rC,
         if opts.DegreeLimit === infinity then {} else
```

**Comment by mahrud** [comment]
Mike, is this done? The tests have passed. GitHub is not happy with your last merge commit, which would be resolved if you rebased and force pushed, but we can also just merge this without rebasing.

**Comment by mikestillman** [comment]
Yes, I'll accept it, but use a merge pull request.

**Pull request merged**